src/main/java/us/kilroyrobotics/subsystems/Elevator.java

```diff
@@ -7,27 +7,130 @@
 import static edu.wpi.first.units.Units.Inches;
 import static edu.wpi.first.units.Units.Meters;
 
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.sim.SparkMaxSim;
+import com.revrobotics.spark.SparkBase.ControlType;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkLowLevel.MotorType;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.config.ClosedLoopConfig.FeedbackSensor;
+import com.revrobotics.spark.config.SparkMaxConfig;
 import edu.wpi.first.epilogue.Logged;
 import edu.wpi.first.math.geometry.Pose3d;
 import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.units.measure.Distance;
+import edu.wpi.first.wpilibj.simulation.BatterySim;
+import edu.wpi.first.wpilibj.simulation.ElevatorSim;
+import edu.wpi.first.wpilibj.simulation.RoboRioSim;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import us.kilroyrobotics.Constants.ElevatorConstants;
+import us.kilroyrobotics.Constants.SimulationConstants;
 
 public class Elevator extends SubsystemBase {
+    private SparkMax m_leadMotor;
+    private SparkMax m_followerMotor;
+    private RelativeEncoder m_encoder;
+    private SparkClosedLoopController m_pidController;
+
+    /* Sim Specific */
+    private DCMotor m_simElevatorGearbox;
+    private SparkMaxSim m_simLeadMotor;
+    private ElevatorSim m_simElevator;
+
     /** Creates a new Elevator. */
-    public Elevator() {}
+    public Elevator() {
+        this.m_leadMotor = new SparkMax(ElevatorConstants.kLeftMotorId, MotorType.kBrushless);
+        this.m_followerMotor = new SparkMax(ElevatorConstants.kRightMotorId, MotorType.kBrushless);
+        this.m_pidController = this.m_leadMotor.getClosedLoopController();
+        this.m_encoder = this.m_leadMotor.getEncoder();
+
+        // Configure
+        SparkMaxConfig leadMotorConfig = new SparkMaxConfig();
+        leadMotorConfig
+                .closedLoop
+                .feedbackSensor(FeedbackSensor.kPrimaryEncoder)
+                .pidf(5.0, 0.0, 0.0, 0.0);
```

Comment:
These need to be extracted out to the constants file

Reply:
done